Assets/MixedRealityToolkit/_Core/DataProviders/SpatialObservers/BaseMixedRealitySpatialObserverDataProvider.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.DataProviders.SpatialObservers;
+using Microsoft.MixedReality.Toolkit.Core.Services;
+using System.Collections;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.DataProviders.SpatialObservers
+{
+    /// <summary>
+    /// Base <see cref="IMixedRealitySpatialObserverDataProvider"/> implementation
+    /// </summary>
+    public abstract class BaseMixedRealitySpatialObserverDataProvider : BaseDataProvider, IMixedRealitySpatialObserverDataProvider
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="priority"></param>
+        /// <param name="profile"></param>
+        protected BaseMixedRealitySpatialObserverDataProvider(string name, uint priority, BaseMixedRealitySpatialObserverProfile profile) : base(name, priority)
+        {
+            if (profile == null)
+            {
+                Debug.LogError($"Missing profile for {name}");
+                return;
+            }
+
+            if (MixedRealityToolkit.SpatialAwarenessSystem != null)
+            {
+                SourceId = MixedRealityToolkit.SpatialAwarenessSystem.GenerateNewObserverId();
+            }
+
+            StartupBehavior = profile.StartupBehavior;
+            UpdateInterval = profile.UpdateInterval;
+            PhysicsLayer = profile.PhysicsLayer;
```

Comment:
This _does_ support multiple forms of spatial data. 

This is the base class to support those multiple forms. Each type derives from this base class and has its own configuration settings.

Reply:
> Please move this property to the specific observer types as it breaks previously established behavior.

No, it belongs here.

Reply:
As defined, a single observer that reports both meshes and planes, for example, can only support a single physics layer. The desire is to enable a single observer that can manage multiple forms.

Reply:
Mesh and surface observers should be different data providers, (and was treated as such in the "current" definition).

The fact that it only reports a single physics layer is intentional as the observer keeps track of its own generated GameObjects.

Surface observers will create and have their own meshes on physics layers defined by the `IMixedRealitySpatialObserverDataModel`

Reply:
> Mesh and surface observers should be different data providers, (and was treated as such in the "current" definition).

Not necessarilly. There are scenarios where a single data provider needs to be responsible for both. Requiring the data to be separated into multiple observers would lead to duplicated code and potentially out of sync data.,